### ISSUE TITLE
Four new important commands

### ISSUE THREAD

**Issue by Dariosky-01** [TheJoeSmo/Foundry#193]
Salutations
Often there is a need to use the "select all" commands; "copy"; "cut" ; "paste"
I thought that they could very well be part of the equipment of the buttons above and of one of the menus, (although I would make a new one dedicated to the work area)
Because they are often useful to avoid going to look for objects and or enemies inside the left internal window where the icons are placed, it is easier to work having these buttons at the top if I have to move a part of the level, or a group of objects or enemies, sometimes it happens just to have to make copies of the entire level for adjustments in this case the "select all" button is very useful to use it together with the copy button and then with the paste one.
Sometimes it is necessary to use cut because a piece of layer, a group of objects or even a single object or enemy you want to remove without necessarily having to drag it with the pointer and select paste by first indicating the point with a click.
I think that the paste button, however, should work as a switch by remaining pressed until the click inside the editor so it would be easier to understand where to apply everything, it is obvious that before the click it would be interesting to see "the ghost of objects" move with the pointer waiting for the click to click the "paste" command and allow you to auto-release the "paste" button (without seeing it pressed again).

**Comment by TheJoeSmo** [comment]
These options already exist as keyboard shortcuts.  Can you provide an image of where they ought to exist as physical buttons?

**Comment by Dariosky-01** [comment]
![IMG_20220419_063705](https://user-images.githubusercontent.com/81451904/163921305-eda2733a-175d-4609-b555-d31ec2f15ad0.jpg)

Here.
Left to right in this sequence:
1-Select all
2-Cut
3-Copy
4-Paste
.


I would like to add that the ctrl button could as always be used in this case to block the functionality in adding elements, if for example you left the ctrl pressed while pressing cut, this would have to block the cut button making it appear as always pressed until I release ctrl to make different cuts or to leave paste pressed, for different gluing of the same thing, it can also work for the copy function in the same way, you would not see what it stores in the memory but if you see that the button is pressed you know that it would continue to memorize the elements by adding them to the memory with each click of the left mouse button, always with the ctrl pressed.

**Comment by TheJoeSmo** [comment]
I believe select all should go in the level menu.  It is unrealistic that one would select all of the objects at once.  And if one wanted to, they could already use Ctrl-A or use the side menu.  Instead I propose that it goes inside the level menu unless you can provide a use case where it'd be used often.

I agree with 2-4.

**Comment by Dariosky-01** [comment]
Joe the copy button is used for example to avoid inconvenient dragging at times with old mice, so it must not be ignored, made inaccessible or even inaccessible among the important buttons at the top, because copying an element already present in the workspace , like an enemy, if you click for another enemy with the paste command (or many clicks of the paste command by simply holding down ctrl) it is much easier to add enemies or different objects if the copied ones are objects.

True, the "select all" command is not widely used, but for those unfamiliar with shortcuts it is certainly much more useful.

However, I recommend that when you press the new button paste the selected object or enemy (or multiple objects copied with copy or cut) before applying them in the layer see them next to the pointer as ghosts on its right drawn as a whole have been copied or cut in the layer, so that the lower left corner of the general drawing all together they create (with a transparent background) coincides with the pointer at the point where the click acts.

**Comment by TheJoeSmo** [comment]
I am not arguing against the copy and undo buttons.

My question is in regards to the select all button.  You did not address my question of why this is a critical button.  I believe it isn't.  Perhaps it is useless, but you can fundamentally do everything without a select all button.  There are other menus that allow you to select all the items, even with no knowledge of their location in game.  Perhaps it is slower or less efficient, but all alternatives mustn't be the best.  That is why I argue it ought be placed out of the primary functionality of the editor.  If we place too many elements of little value into the editor, it clutters up the space, which is poor taste.

**Comment by Dariosky-01** [comment]
In the last line of your previous comment it says

"I agree with 2-4"

This made me understand you only agreed with the "cut" and "paste" button because the numbered points I used for these buttons are just 2 and 4, that's why I answered that way.

**Comment by TheJoeSmo** [comment]
Fair enough.  Unless there are any other opinions on the matter, we will put the 2-4 as an option on the main tab and put the `select all` button inside the level menu.
